pkg/services/services.go

```diff
@@ -41,7 +41,29 @@ func toContainerLabelsService(service *v1.ServiceInstance) (result []kclient.Obj
 	return
 }
 
-func toContainerService(service *v1.ServiceInstance) (result []kclient.Object) {
+func toContainerService(ctx context.Context, c kclient.Client, service *v1.ServiceInstance) (result []kclient.Object, err error) {
+	svcPorts := ports.ToServicePorts(service.Spec.Ports)
+
+	// Check whether this is the main ServiceInstance for this container.
+	if service.Spec.Container != service.Name {
+		// Return an error if there are any non-HTTP ports defined. This could maybe cause problems with Istio.
+		for _, port := range service.Spec.Ports {
+			if port.Protocol != v1.ProtocolHTTP {
+				return nil, fmt.Errorf("container service %s has non-HTTP port %d\nservices defined for existing containers must contain only HTTP ports", service.Name, port.Port)
+			}
+		}
+
+		// Get the main ServiceInstance for this container.
+		mainService := &v1.ServiceInstance{}
+		if err = c.Get(ctx, kclient.ObjectKey{Name: service.Spec.Container, Namespace: service.Namespace}, mainService); err != nil {
+			return
+		}
```

Comment:
Are we guaranteed that such a service exists? Should we handle `IsNotFound` errors here?

Reply:
I believe it is guaranteed, yes. This still feels like an awkward assumption to make though, so I will add in some logic to ignore the `IsNotFound` error in case it comes up.